libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -857,4 +857,180 @@ async_contract_fulfillment memcpy_async(void * __destination, void const * __sou
 
 _LIBCUDACXX_END_NAMESPACE_CUDA
 
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA_EXPERIMENTAL
+
+// - cp_async_bulk_global_to_shared
+// - cp_async_bulk_shared_to_global
+// - cp_async_bulk_tensor_{1,2,3,4,5}d_global_to_shared
+// - cp_async_bulk_tensor_{1,2,3,4,5}d_shared_to_global
+// - fence_proxy_async_shared_cta
+// - cp_async_bulk_commit_group
+// - cp_async_bulk_wait_group_read<0, …, 7>
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_global_to_shared(void *dest, const void *src, uint32_t size, cuda::barrier<cuda::thread_scope_block> &bar)
+{
+#if (_LIBCUDACXX_DEBUG_LEVEL >= 2)
+    _LIBCUDACXX_DEBUG_ASSERT(size % 16 == 0);
+    _LIBCUDACXX_DEBUG_ASSERT(__isShared(dest));
+    _LIBCUDACXX_DEBUG_ASSERT(__isGlobal(src));
+#endif
+
+  asm volatile(
+    "cp.async.bulk.shared::cluster.global.mbarrier::complete_tx::bytes [%0], [%1], %2, [%3];\n"
+    :
+    : "r"(static_cast<uint32_t>(__cvta_generic_to_shared(dest))),
+      "l"(static_cast<uint64_t>(__cvta_generic_to_global(src))),
+      "r"(size),
+      "r"(static_cast<uint32_t>(__cvta_generic_to_shared(cuda::device::barrier_native_handle(bar))))
+    : "memory");
+}
+
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_shared_to_global(void *dest, const void * src, uint32_t size)
+{
+  asm volatile(
+    "cp.async.bulk.global.shared::cta.bulk_group [%0], [%1], %2;\n"
+    :
+    : "l"(static_cast<uint64_t>(__cvta_generic_to_global(dest))),
+      "r"(static_cast<uint32_t>(__cvta_generic_to_shared(src))),
+      "r"(size)
+    : "memory");
+}
+
+// https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#data-movement-and-conversion-instructions-cp-async-bulk-tensor
+_LIBCUDACXX_DEVICE
+void cp_async_bulk_tensor_1d_global_to_shared(
+    void *dest, const void *tensor_map , int c0, cuda::barrier<cuda::thread_scope_block> &bar)
```

Comment:
The type of `tensor_map` is actually `CUtensorMap`. We could just forward declare the type instead of using a `void*`. 

This may be slightly more convenient for readability and maintainability.